File: dotnet/Src/ButterflyNet.Parser/Scopes/LinkScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class LinkScope : InlineScope {
		public LinkScope(string url, string baseUrl) {
			Url = url;
			BaseUrl = baseUrl;
		}

		public string Url { get; private set; }
		public string BaseUrl { get; private set; }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenLink(Url, BaseUrl);
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseLink();
		}
	}
}

File: src/js/src/scopes/LinkScope.js
function LinkScope(url, baseUrl) {
	this.url = url;
	this.baseUrl = baseUrl === undefined ? "/" : baseUrl;
	
	this.type = ScopeTypeCache.link;
	this.level = ScopeLevel.inline;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openLink(this.url, this.baseUrl);
	};
	this.close = function(analyzer) {
		analyzer.closeLink();
	};
}

extend(Scope, LinkScope);